test/evp_extra_test.c

```diff
@@ -3540,6 +3541,8 @@ static int test_empty_salt_info_HKDF(void)
 
     if (!TEST_int_gt(EVP_PKEY_derive_init(pctx), 0)
             || !TEST_int_gt(EVP_PKEY_CTX_set_hkdf_md(pctx, EVP_sha256()), 0)
+            || !TEST_int_gt(EVP_PKEY_CTX_set1_hkdf_salt(pctx, fake,
+                                                        sizeof(fake) - 1), 0)
```

Comment:
not really overjoyed with how this does sizeof(salt) - 1 and sizeof(info)-1..
These are just 0 values.

Where is the test for setting salt to NULL?